src/systemd-sonic-generator/systemd-sonic-generator.c

```diff
@@ -0,0 +1,264 @@
+#include <assert.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <linux/limits.h>
+
+#define MAX_NUM_TARGETS 5
+#define MAX_NUM_UNITS 128
+
+static const char* UNIT_FILE_PREFIX = "/etc/systemd/system/";
+static const char* CONFIG_FILE = "/etc/sonic/generated_services.conf";
+
+
+void strip_trailing_newline(char* str) {
+    /***
+    Strips trailing newline from a string if it exists
+    ***/
+
+    size_t l = strlen(str);
+    if (l > 0 && str[l-1] == '\n')
+        str[l-1] = '\0';
+}
+
+
+static int get_target_lines(char* unit_file, char* target_lines[]) {
+    /***
+    Gets installation information for a given unit file
+
+    Returns lines in the [Install] section of a unit file
+    ***/
+    FILE *fp;
+    char *line = NULL;
+    size_t len = 0;
+    ssize_t read;
```

Comment:
Please rename read to something else (usually nread).
read is a function from a standard library.